src/main/java/core/basesyntax/service/impl/TransactionParserImpl.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Transaction;
+import core.basesyntax.service.TransactionParser;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TransactionParserImpl implements TransactionParser {
+    private static final String SEPARATOR = ",";
+    private static final int EXPECTED_PARTS = 3;
+    private static final int OPERATION_TYPE = 0;
+    private static final int FRUIT_NAME = 1;
+    private static final int QUANTITY = 2;
+
+    @Override
+    public List<Transaction> parse(List<String> rawData) {
+        if (rawData == null || rawData.isEmpty()) {
+            throw new IllegalArgumentException("Invalid transaction line: " + rawData);
```

Comment:
The error message for an empty or null `rawData` should be more informative. It should not include the `rawData` itself, but instead explain the reason why it is considered invalid, e.g., 'Provided data for parsing is null or empty, which is not allowed.'

Reply:
the logic was changed